src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1232,6 +1239,37 @@ def _get_log(url, user_name, password, log_file=None):
                 if not data:
                     break
                 f.write(data)
+    elif number_of_lines:  # print last 'n' number of lines
+        api_str = r.read().decode(encoding='utf-8')
+
+        href_count = api_str.count('https')
+
+        if href_count > 1:
+            logger.info('Will show last %s lines of logs per instance', number_of_lines)
+
+        for x in range(0, href_count):  # pylint: disable=unused-variable
+            index_of_path = api_str.index('\",\"path')
+            href_str = api_str[api_str.index("https"):index_of_path]
+            size = api_str[api_str.index('\"size\":') + len('\"size\":'):api_str.index(',\"href\"')]
+            logger.info('\nUsing endpoint: %s', href_str)
+            url = href_str
+            if int(size) > 512000:
+                logger.warning('Size of logs:%s bytes, will fetch only last 512000 bytes', size)
```

Comment:
Is it a server side limitation? I think we should only emit a warning if the line number supplied is more than 512000 can cover.

Reply:
It is not, but this log file can grow arbitrarily so we need to set the cap somewhere. My team agreed alongside with folks from CSM that 512k is a reasonable cap. Also I'll make the change to emit the warning only in the case the line number is more than what 512k can supply.